### PR TITLE
remove icons/* entrypoints

### PR DESCRIPTION
**Pull request by sachinraja**
Remove the `icons/*.js` entry points. Require that users instead use the `simple-icons/icons` entry point to import all icons.

### CODE REVIEW

**Comment by ericcornelissen** [comment]
Should we push this change out to v7.0.0 and allow users some more time to update their code  to the new way of importing individual icons? The new way will only be available starting from the next release, less than a month before the v6.0.0 release.

**Comment by sachinraja** [comment]
> allow users some more time to update their code to the new way of importing individual icons

There is nothing pushing current users to use the new imports though. I see no reason why they would change to use them without us removing the entry points. They may not even know about it, I doubt they are reading the changelog on every `simple-icons` release. There is also nothing truly problematic with staying on an older version until the imports are fixed, which should not take very long.

**Comment by ericcornelissen** [comment]
> There is nothing pushing current users to use the new imports though. I see no reason why they would change to use them without us removing the entry points. They may not even know about it, I doubt they are reading the changelog on every simple-icons release. 

Which is exactly why I **think** we should provide package users some time to migrate.

Also, we can do more than put it in the changelog. We can add JSDocs and the `@deprecated` tag to these exports so that IDEs can recognize those imports are deprecated.

But the keyword here is **think**. Ultimately, whether you want to provide your package users some transition time for breaking changes is a choice and if other maintainers think that's not necessary than we can go ahead and merge this in before v6.


> There is also nothing truly problematic with staying on an older version until the imports are fixed, which should not take very long.

I can agree with that, to an extends, for this package (because it has 0 dependencies). However, it still prevents users from getting the latest (new and updated) icons.

Also, similarly, there's nothing problematic about providing deprecated functionality for 6 more months.

**Comment by sachinraja** [comment]
> Also, we can do more than put it in the changelog. We can add JSDocs and the @deprecated tag to these exports so that IDEs can recognize those exports are deprecated.

I see your point there, but I feel that is unnecessary. Major releases are supposed to contain breaking changes. I think that migrations are expected.

> However, it still prevents users from getting the latest (new and updated) icons.

My thinking was that if a user wanted a new icon, they could just migrate to the new imports, which wouldn't take longer than a few minutes.

Mentioning @simple-icons/maintainers for more input

**Comment by sachinraja** [comment]
@mondeja @jorgeamadosoria do you have an opinion on this?

**Comment by mondeja** [comment]
> We can add JSDocs and the @deprecated tag to these exports so that IDEs can recognize those exports are deprecated.

I like this idea. We should warning users using the old imports to make the migration, and remove them in v8.0.0.

**Comment by sachinraja** [comment]
Ok, I'm fine with delaying this until v7.0.0 then. I'll work on adding the warnings.

**Comment by mondeja** [comment]
> Ok, I'm fine with delaying this until v7.0.0 then. I'll work on adding the warnings.

Would you mind to open a PR for that before this sunday, when the v6 release is planned @sachinraja?

**Comment by sachinraja** [comment]
I'm actually working on it right now, so yes, I'll definitely have it open by then!

**Comment by mondeja** [comment]
## Important NOTE ⚠️ 

The `console.warn` tests mock added in #6900 (see [these lines](https://github.com/simple-icons/simple-icons/pull/6900/files#diff-927d3c3a74ad4643c7a9ec975a87af3df483927c1280bdf10206e29e6fa14b45R5-R6)) must be removed in this PR before merge it!

**Comment by sachinraja** [comment]
That whole file would actually be removed

**Comment by jorgeamadosoria** [comment]
> @mondeja @jorgeamadosoria do you have an opinion on this?

I agree that major releases, in theory, should be expected to have breaking changes. Having said that, if we can avoid them that's always preferable. So I agree with @mondeja's review on this one.

**Comment by sachinraja** [comment]
@sisp Feel free to review if you have time. This is a breaking change for the next major (May 28).

**Review by mondeja — CHANGES_REQUESTED** [review]
Additionally:

- We don't need [this export](https://github.com/simple-icons/simple-icons/blob/5ece738699b42febdf6155f4c66adebad1a0f7e2/package.json#L22) right?
- We must change the `clean script` to remove [`icons/*.js icons/*.d.ts`](https://github.com/simple-icons/simple-icons/blob/5ece738699b42febdf6155f4c66adebad1a0f7e2/package.json#L60).
- [This line in *.dockerignore*](https://github.com/simple-icons/simple-icons/blob/5ece738699b42febdf6155f4c66adebad1a0f7e2/.dockerignore#L6) it not needed. Must be replaced by other generated files: `/icons.js`, `/icons.mjs` and `/icons.d.ts`.
- [`!alias.d.ts` line](https://github.com/simple-icons/simple-icons/blob/5ece738699b42febdf6155f4c66adebad1a0f7e2/.npmignore#L16) should be removed from *.npmignore*.
- [These two lines](https://github.com/simple-icons/simple-icons/blob/5ece738699b42febdf6155f4c66adebad1a0f7e2/.prettierignore#L13-L14) should be removed from *.prettierignore*.

File: README.md
Comment (sachinraja) [reply README.md:77]: to be honest, I would really prefer we remove the `simpleIcons`/`index.js` entry point (see #6789)

Comment (mondeja) [reply README.md:77]: I agree, but meanwhile this part is a bit confusing.

**Review by mondeja — APPROVED** [review]

**Comment by mondeja** [comment]
Let's go with this!

**Pull request merged**
